### DISCUSSION TITLE
Cross compilation linux -> windows

### DISCUSSION THREAD

**Discussion by lexxvir** [slint-ui/slint#2471] [Q&A]
Hi,

I have tried to cross compile the gallery demo for Windows in the linux host (ubuntu 22.04) using opengl backend.
Compilation have finished successfully with following command: `cargo run --target x86_64-pc-windows-gnu`
I able to run application under Wine.

But when I run application in Windows 11 there is no text at all:

![image](https://user-images.githubusercontent.com/331063/229592568-1d58e5fc-8876-44ef-abac-e76fa935fec7.png)

I have tried both release 1.0.0 and master branch.

**Comment by tronical** [comment]
This is most likely cause by a bad OpenGL driver. Unfortunately cross compiling the Skia backend won't be possible with mingw. You could try using the software renderer - although you might run into some missing features related to text. Try passing `--features slint/renderer-winit-software` and set `SLINT_BACKEND=winit-software` as environment variable when running.

**Reply by tronical** [reply]
@lexxvir btw, I'm curious: What graphics card & drivers are you using on your Windows installation? I haven't been able to reproduce this issue yet, but I've seen it. I'm interesting in finding out how to reproduce it, in order to assess if there are any easy workarounds.

**Reply by lexxvir** [reply]
Hi @tronical !

Thank you for the suggestion about software render. I have tried it but without luck. Issue the same - text is not rendered at all.
I suppose that some fonts are missing probably...

Regarding my env: it is Windows 11 Visual Studio development image for VirtualBox 6.1 that I run in Ubuntu 22.04.
Also, I have tried the same application in real Windows 11 and it works fine!

So probably it is issue with VirtualBox emulation.

**Reply by tronical** [reply]
Ahh, thank you! VirtualBox rings a bell. That could be it, and that could be the key to reproducing it.

**Reply by lexxvir** [reply]
So, I installed recent VirtualBox version 7.0 and now it works! Thank you for the help!
